seatunnel-connectors-v2/connector-amazondynamodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/amazondynamodb/sink/AmazondynamodbWriter.java

```diff
@@ -0,0 +1,173 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.seatunnel.connectors.seatunnel.amazondynamodb.sink;
+
+import org.apache.seatunnel.api.table.type.ArrayType;
+import org.apache.seatunnel.api.table.type.BasicType;
+import org.apache.seatunnel.api.table.type.MapType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.connectors.seatunnel.amazondynamodb.config.AmazondynamodbSourceOptions;
+import org.apache.seatunnel.connectors.seatunnel.common.sink.AbstractSinkWriter;
+
+import software.amazon.awssdk.auth.credentials.AwsBasicCredentials;
+import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider;
+import software.amazon.awssdk.core.SdkBytes;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient;
+import software.amazon.awssdk.services.dynamodb.model.AttributeValue;
+import software.amazon.awssdk.services.dynamodb.model.PutItemRequest;
+
+import java.io.IOException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class AmazondynamodbWriter extends AbstractSinkWriter<SeaTunnelRow, Void> {
+
+    private final AmazondynamodbSourceOptions amazondynamodbSourceOptions;
+    private final SeaTunnelRowType seaTunnelRowType;
+    private final DynamoDbClient dynamoDbClient;
+    private final List<AttributeValue.Type> measurementsType;
+
+    public AmazondynamodbWriter(AmazondynamodbSourceOptions amazondynamodbSourceOptions, SeaTunnelRowType seaTunnelRowType) {
+        this.amazondynamodbSourceOptions = amazondynamodbSourceOptions;
+        this.seaTunnelRowType = seaTunnelRowType;
+        dynamoDbClient = DynamoDbClient.builder()
+            .endpointOverride(URI.create(amazondynamodbSourceOptions.getUrl()))
+            // The region is meaningless for local DynamoDb but required for client builder validation
+            .region(Region.of(amazondynamodbSourceOptions.getRegion()))
+            .credentialsProvider(StaticCredentialsProvider.create(
+                AwsBasicCredentials.create(amazondynamodbSourceOptions.getAccessKeyId(), amazondynamodbSourceOptions.getSecretAccessKey())))
+            .build();
+        this.measurementsType = convertTypes(seaTunnelRowType);
+    }
+
+    @Override
+    public void write(SeaTunnelRow element) throws IOException {
+
+        dynamoDbClient.putItem(convertRow(element, seaTunnelRowType));
```

Comment:
Is putItem a synchronous write, can I use batch write to improve performance

Reply:
> Is putItem a synchronous write, can I use batch write to improve performance



> Is putItem a synchronous write, can I use batch write to improve performance

Done.